src/main/java/spireMapOverhaul/zones/ambushed/AmbushedZone.java

```diff
@@ -0,0 +1,380 @@
+        List<String> filteredCardIDs = getFilteredCardIDsForShop(rarity, type, isColored, selectedCardIDs);
+
+        // Fallback 1: If no cards of the specified type, try another type
+        if (filteredCardIDs.isEmpty()) {
+            for (AbstractCard.CardType fallbackType : AbstractCard.CardType.values()) {
+                if (fallbackType != type) {
+                    filteredCardIDs = getFilteredCardIDsForShop(rarity, fallbackType, isColored, selectedCardIDs);
+                    if (!filteredCardIDs.isEmpty()) break;
+                }
+            }
+        }
+
+        // Fallback 2: If still no cards, try a different rarity
+        if (filteredCardIDs.isEmpty()) {
+            for (AbstractCard.CardRarity fallbackRarity : AbstractCard.CardRarity.values()) {
+                if (fallbackRarity != rarity) {
+                    filteredCardIDs = getFilteredCardIDsForShop(fallbackRarity, type, isColored, selectedCardIDs);
+                    if (!filteredCardIDs.isEmpty()) break;
+                }
+            }
+        }
+
+        if (filteredCardIDs.isEmpty()) {
+            return null;
+        }
+
+        String selectedCardID = filteredCardIDs.get(AbstractDungeon.cardRng.random(filteredCardIDs.size() - 1));
+        selectedCardIDs.add(selectedCardID);
+
+        return CardLibrary.getCard(selectedCardID).makeCopy();
+    }
+
+
+    private List<String> getFilteredCardIDsForShop(AbstractCard.CardRarity rarity, AbstractCard.CardType type, boolean isColored, HashSet<String> selectedCardIDs) {
+        // Filter your card draw cards by rarity, type, and whether it's colored or colorless
+        return cardDrawCardIDs.stream()
+                .filter(id -> isCardForShop(id, rarity, type, isColored))
+                .filter(id -> !selectedCardIDs.contains(id)) // Exclude already selected card IDs
+                .collect(Collectors.toList());
+    }
+
+
+    private boolean isCardForShop(String cardID, AbstractCard.CardRarity rarity, AbstractCard.CardType type, boolean isColored) {
+        AbstractCard card = CardLibrary.getCard(cardID);
+        boolean isCardColored = card.color != AbstractCard.CardColor.COLORLESS;
+        return card != null && card.rarity == rarity && card.type == type && isCardColored == isColored;
+    }
+
+    @Override
+    public void postCreateShopPotions(ShopScreen screen, ArrayList<StorePotion> potions) {
+        if (potions.size() >= 3) {
+            potions.clear();
+
+            // Add Swift Potion with its appropriate price
+            potions.add(new StorePotion(PotionHelper.getPotion(SwiftPotion.POTION_ID), 0, screen));
+
+            // Add Gambler's Brew with its appropriate price
+            potions.add(new StorePotion(PotionHelper.getPotion(GamblersBrew.POTION_ID), 1, screen));
+
+            // Add Snecko Oil with its appropriate price
+            potions.add(new StorePotion(PotionHelper.getPotion(SneckoOil.POTION_ID), 2, screen));
+        }
+    }
```

Comment:
Try adding the following to get the normal price variation for these potions:
```

        for (StorePotion potion: potions) {
            if (!Settings.isDailyRun) {
                potion.price = MathUtils.round(potion.price * AbstractDungeon.merchantRng.random(0.95F, 1.05F));
            }
        }
```

Reply:
I did this myself